apps/omg_watcher_info/test/support/factory.ex

```diff
@@ -98,21 +98,28 @@ defmodule OMG.WatcherInfo.Factory do
   @doc """
   Txoutput factory.
 
-  Generates a txoutput. The txindex, oindex and spending_tx_oindex are defaulted to 0.
-  These two values need to be overridden depending on the transaction you aim to build.
+  Generates a txoutput. A new block and transaction are generated per each txoutput built.
+  Therefore, if you are overriding some values, also consider its relation to other values. E.g:
+
+    - To override `blknum`, also consider overriding `txindex`.
+    - To override `creating_transaction`, also consider overriding `txindex` and `oindex`.
+    - To override `spending_transaction`, also consider overriding `spending_tx_oindex`
   """
   def txoutput_factory() do
+    block = build(:block)
+    transaction = build(:transaction, blknum: block.blknum)
```

Comment:
Because of this mapping in `db/transaction.ex`
`belongs_to(:block, DB.Block, foreign_key: :blknum, references: :blknum, type: :integer)`

I think `transaction = build(:transaction, block: block)` is a more natural and consistent use of the factories rather than `blknum: block.blknum`

You decide what to do for you now, it's not blocking.

Reply:
Yup this code did fail.